mifosng-android/src/main/java/com/mifos/api/datamanager/DataManagerSearch.java

```diff
@@ -17,14 +25,55 @@
 public class DataManagerSearch {
 
     public final BaseApiManager baseApiManager;
+    public final DatabaseHelperClient mDatabaseHelperClient;
+    public final DatabaseHelperGroups mDatabaseHelperGroup;
+    public final DatabaseHelperCenter mDatabaseHelperCenter;
+    public final DatabaseHelperLoan mDatabaseHelperLoan;
+    public final DatabaseHelperSavings mDatabaseHelperSaving;
 
     @Inject
-    public DataManagerSearch(BaseApiManager baseApiManager) {
+    public DataManagerSearch(BaseApiManager baseApiManager,
+                             DatabaseHelperClient databaseHelperClient,
+                             DatabaseHelperGroups databaseHelperGroup,
+                             DatabaseHelperCenter databaseHelperCenter,
+                             DatabaseHelperLoan databaseHelperLoan,
+                             DatabaseHelperSavings databaseHelperSaving) {
         this.baseApiManager = baseApiManager;
+        mDatabaseHelperClient = databaseHelperClient;
+        mDatabaseHelperGroup = databaseHelperGroup;
+        mDatabaseHelperCenter = databaseHelperCenter;
+        mDatabaseHelperLoan = databaseHelperLoan;
+        mDatabaseHelperSaving = databaseHelperSaving;
     }
 
-    public Observable<List<SearchedEntity>> searchResources(String query, String resources,
-                                                            Boolean exactMatch) {
-        return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+    public Observable<List<SearchedEntity>> searchResources(final String query, String resources,
+                                                            final Boolean exactMatch) {
+        switch (PrefManager.getUserStatus()) {
+            case 0:
+                return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+            case 1:
+                if (resources.equals(Constants.ENTITY_TYPE_CLIENTS)) {
+                    return mDatabaseHelperClient.readAllClientsByQuery(query, exactMatch);
+                } else if (resources.equals(Constants.ENTITY_TYPE_GROUPS)) {
+                    return
+                            mDatabaseHelperGroup.readAllGroupsByQuery(query, exactMatch)
+                                    .mergeWith(mDatabaseHelperCenter.readAllCentersByQuery(query,
```

Comment:
@therajanmaurya @puneetkohli521 The app is crashing because of this "mergeWith" method. I have tried everything from my side but I am not able to resolve this issue.
Here is error log - 
 java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
                                                                           at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:62)
                                                                           at android.os.Handler.handleCallback(Handler.java:751)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                           at android.os.Looper.loop(Looper.java:154)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6123)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
                                                                        Caused by: rx.exceptions.OnErrorFailedException: Error occurred when trying to propagate error to Observer.onError
                                                                           at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:192)
                                                                           at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:120)
                                                                           at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:204)
                                                                           at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:144)
                                                                           at rx.internal.operators.OperatorSubscribeOn$1$1.onNext(OperatorSubscribeOn.java:53)
                                                                           at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call(OperatorObserveOn.java:215)
                                                                           at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
                                                                           at android.os.Handler.handleCallback(Handler.java:751) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                                           at android.os.Looper.loop(Looper.java:154) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6123) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) 
                                                                        Caused by: rx.exceptions.CompositeException: 2 exceptions occurred. 
                                                                           at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:192) 
                                                                           at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:120) 
                                                                           at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:204) 
                                                                           at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:144) 
                                                                           at rx.internal.operators.OperatorSubscribeOn$1$1.onNext(OperatorSubscribeOn.java:53) 
                                                                           at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call(OperatorObserveOn.java:215) 
                                                                           at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55) 
                                                                           at android.os.Handler.handleCallback(Handler.java:751) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                                           at android.os.Looper.loop(Looper.java:154) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6123) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757) 
                                                                        Caused by: rx.exceptions.CompositeException$CompositeExceptionCausalChain: Chain of Causes for CompositeException In Order Received =>
                                                                           at com.crashlytics.android.core.TrimmedThrowableData.<init>(TrimmedThrowableData.java:19)
                                                                           at com.crashlytics.android.core.TrimmedThrowableData.<init>(TrimmedThrowableData.java:20)
                                                                           at com.crashlytics.android.core.TrimmedThrowableData.<init>(TrimmedThrowableData.java:20)
                                                                           at com.crashlytics.android.core.CrashlyticsController.writeSessionEvent(CrashlyticsController.java:1068)
                                                                           at com.crashlytics.android.core.CrashlyticsController.writeFatal(CrashlyticsController.java:831)
                                                                           at com.crashlytics.android.core.CrashlyticsController.access$200(CrashlyticsController.java:59)
                                                                           at com.crashlytics.android.core.CrashlyticsController$6.call(CrashlyticsController.java:285)
                                                                           at com.crashlytics.android.core.CrashlyticsController$6.call(CrashlyticsController.java:278)
                                                                           at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                                           at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                           at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                           at io.fabric.sdk.android.services.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:75)
                                                                           at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:30)
                                                                           at java.lang.Thread.run(Thread.java:761)
                                                                        Caused by: java.lang.IllegalArgumentException: Counter has been corrupted!
                                                                           at com.mifos.utils.SimpleCountingIdlingResource.decrement(SimpleCountingIdlingResource.java:95)
                                                                           at com.mifos.utils.EspressoIdlingResource.decrement(EspressoIdlingResource.java:37)
                                                                           at com.mifos.mifosxdroid.online.search.SearchFragment.showSearchedResources(SearchFragment.java:129)
                                                                           at com.mifos.mifosxdroid.online.search.SearchPresenter$1.onNext(SearchPresenter.java:65)
                                                                           at com.mifos.mifosxdroid.online.search.SearchPresenter$1.onNext(SearchPresenter.java:48)
                                                                           at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:139)
                                                                           at rx.internal.operators.OperatorSubscribeOn$1$1.onNext(OperatorSubscribeOn.java:53)
                                                                           at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.call(OperatorObserveOn.java:215)
                                                                           at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
                                                                           at android.os.Handler.handleCallback(Handler.java:751)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                           at android.os.Looper.loop(Looper.java:154)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6123)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
08-05 16:32:07.465 30227-30227/com.mifos.mifosxdroid E/AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Counter has been corrupted!
                                                                           at com.mifos.utils.SimpleCountingIdlingResource.decrement(SimpleCountingIdlingResource.java:95)
                                                                           at com.mifos.utils.EspressoIdlingResource.decrement(EspressoIdlingResource.java:37)
                                                                           at com.mifos.mifosxdroid.online.search.SearchFragment.showMessage(SearchFragment.java:141)
                                                                           at com.mifos.mifosxdroid.online.search.SearchPresenter$1.onError(SearchPresenter.java:55)
                                                                           at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:157)
                                                                           at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:120)
                                                                           at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:204)
                                                                           at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:144)